File: SnakePhone/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Android.Graphics.Drawables;
using SnakeModel;

namespace SnakeWear
{
    public class Game : View
    {
        private SnakeGame game;
        private Renderer renderer;

        public Game(Context context, int width, int height, int speed, int highscore) : base(context)
        {
            game = new SnakeGame();

            renderer = new Renderer(game, width, height, highscore);

            //This is why I love lambdas. This is all the game steering I need to do.
            Touch += (s, e) => { if (e.Event.Action == MotionEventActions.Up) game.Goto(renderer.GetGameX((int)e.Event.GetX()), renderer.GetGameY((int)e.Event.GetY())); };
            Touch += (s, e) => { if (game.GameOver) GotoMenu(context); };
            Key += (s, keyCode, e) => { if (keyCode == KeyEvent.KEYCODE_BACK) GotoMenu(context); };

            var t = new System.Timers.Timer();
            t.Interval = 700/speed;
            t.Elapsed += (s, e) =>
            {
                game.Update();
                PostInvalidate();
                if (game.GameOver) t.Stop();
            };
            t.Start();
        }

        protected override void OnDraw(Canvas canvas)
        {
            base.OnDraw(canvas);
            renderer.Render(canvas);
        }

        private void GotoMenu(Context context)
        {
            MainActivity host = (MainActivity)context;
            host.ShowMainMenu(game.Score);
        }
    }   
}

File: android/snakewear/wear/src/main/java/se/pontek/snakewear/Game.java
package se.pontek.snakewear;

import android.content.Context;
import android.graphics.Canvas;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;

import java.util.Timer;
import java.util.TimerTask;

import se.pontek.snakewear.model.SnakeGame;

/**
 * Created by Pontus on 2018-01-05.
 */

public class Game extends View {
    private final int MIN_SWIPE_DIST = 100;

    private SnakeGame game;
    private Renderer renderer;
    private boolean exit = false;
    private float lastX;

    public Game(Context context) {
        super(context);
    }

    public Game(Context context, int width, int height, int speed, int highscore) {
        this(context);

        game = new SnakeGame();
        renderer = new Renderer(game, width, height, highscore);

        setOnTouchListener((s, e) -> {
            if (game.isGameOver()) {
                gotoMenu(context);
                return true;
            }

            if (e.getAction() == MotionEvent.ACTION_UP) {
                if (lastX - e.getX() > MIN_SWIPE_DIST) {
                    gotoMenu(context);
                } else {
                    game.changeDirection(
                            renderer.getGameX((int) e.getX()),
                            renderer.getGameY((int) e.getY())
                    );
                }
            } else if (e.getAction() == MotionEvent.ACTION_DOWN) {
                lastX = e.getX();
            }
            s.performClick();
            return true;
        });

        Timer timer = new Timer();
        TimerTask task = new TimerTask() {
            @Override
            public void run() {
                game.update();
                postInvalidate();
                if (game.isGameOver()) {
                    timer.cancel();
                }
            }
        };

        timer.schedule(task, 0, 700 / speed);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        renderer.render(canvas);
    }

    private void gotoMenu(Context context) {
        MainActivity host = (MainActivity) context;
        host.showMainMenu(game.getScore());
    }
}
